### PR TITLE
React to exceptions [ESD-989]

### PR DESCRIPTION
**Pull request by pmiettinen**
Current master Travis build isn't failing on exceptions.

https://swift-nav.atlassian.net/projects/ESD/issues/ESD-989

At least following problems are addressed:
- Exception from malformed YAML wasn't failing the CI build
- pyinstaller INFO message spam hiding WARNINGS and higher
- "win32com module not found" ImportError

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,25 +18,29 @@ matrix:
...
   - rm -rf ~/home/.enthought
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then export PATH="/usr/local/opt/python@2/libexec/bin:$PATH"; fi
   - make deps 
   - pip list
-
-before_script:
-  - "export DISPLAY=:0.0"
   - "export PATH=/usr/local/bin/:./:$PATH"
-  - Xvfb :0.0 -screen 1 800x600x16 &
-  - sleep 3
```
Comment (denniszollo) [inline .travis.yml]: why are we getting rid of running in the virtual screen buffer (basically our only unit test of the console)?

**Comment by denniszollo** [comment]
If we do this, we need to catch every single exception that traits throws during normal operation.  I don't want the console closing suddenly on a user because of bugs in traits or race conditions that we haven't protected against.  For instance, the tabular adapters throw exceptions some times and I've been unable to fix.  Similarly if users try and drag and drop table elements (easy to do by accident), the console shouldn't close.

File: piksi_tools/console/gui_utils.py [hunk 2]
```diff
@@ -68,6 +68,7 @@ def plot_square_axes(plot, xnames, ynames):
     except:  # noqa
         import traceback
         traceback.print_exc()
+        raise
```
Comment (denniszollo) [inline piksi_tools/console/gui_utils.py]: this squaring fails very often in practice, why raise here?

**Comment by denniszollo** [comment]
Okay I get it now, you are trying to make sure there are more failure cases caught with the --error command line argument.   I misread the PR earlier and though it changed behavior without the --error argument.

**Comment by denniszollo** [comment]
> If we do this, we need to catch every single exception that traits throws during normal operation. I don't want the console closing suddenly on a user because of bugs in traits or race conditions that we haven't protected against. For instance, the tabular adapters throw exceptions some times and I've been unable to fix. Similarly if users try and drag and drop table elements (easy to do by accident), the console shouldn't close.

^^ please ignore this one now, given my better understanding of this PR.

File: piksi_tools/console/gui_utils.py (hunk 2 above)
Comment (pmiettinen) [reply piksi_tools/console/gui_utils.py]: This is quite experimental PR, we can revert some parts as necessary if somekind of conclusion is reached.

File: .travis.yml (hunk 1 above)
Comment (pmiettinen) [reply .travis.yml]: With this config we get on linux:

```
Unexpected X11 display depth=8!
There's no Visual compatible with minimal AGG requirements:
At least 15-bit color depth and True- or DirectColor class.
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/enable/qt4/base_window.py", line 198, in paintEvent
    self.handler.paintEvent(event)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/enable/qt4/base_window.py", line 50, in paintEvent
    self._enable_window._paint(event)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/enable/abstract_window.py", line 458, in _paint
    self._gc = self._create_gc(size)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/enable/qt4/image.py", line 23, in _create_gc
    bottom_up = 0)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/kiva/agg/__init__.py", line 40, in __init__
    bottom_up)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/kiva/agg/agg.py", line 797, in __init__
    raise ValueError(msg)
ValueError: Image depth and format are incompatible
```

https://travis-ci.org/swift-nav/piksi_tools/jobs/443883949#L1573

I have configured osx and linux specific setups in `before_install` which get rid of this error.

File: .travis.yml [hunk 3]
```diff
@@ -18,25 +18,29 @@ matrix:
   include:
    - os: linux
      language: python
+     before_install:
+      - "export DISPLAY=:0.0"
+      # set pixel depth https://docs.travis-ci.com/user/gui-and-headless-browsers/
+      - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_00.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :00 -ac -screen 0 1280x1024x16"
```
Comment (pmiettinen) [inline .travis.yml]: TODO: see if `Xvfb :0.0 -screen 1 1280x1024x16 &` works also here

Comment (pmiettinen) [reply .travis.yml]: Not working, giving the color depth error.

**Comment by pmiettinen** [comment]
Few warnings in travis build left:

```
-c: cannot connect to X server 
24750 WARNING: Hidden import "dbm.ndbm" not found!
24750 WARNING: Hidden import "dbm.dumb" not found!
24750 WARNING: Hidden import "dbm.gnu" not found!
27109 WARNING: library Advapi32 required via ctypes not found
27131 WARNING: library setupapi required via ctypes not found
```

- Xserver warning origin and severity currently unknown
- Hidden import warning severity currently unknown
- required via ctypes not found warning apparently harmless, see https://github.com/pyinstaller/pyinstaller/issues/1403

**Comment by pmiettinen** [comment]
Looks like `ImportError: No module named win32com` is back with `pyinstaller-3.4`.

File: .travis.yml [hunk 4]
```diff
@@ -37,8 +37,9 @@ matrix:
         - sudo pip install tox
      before_script:
         - export DISPLAY=:0.0
-        - Xvfb ${DISPLAY} -screen 0 800x600x16 & PID=$!
-        - sleep 3  # give xvfb some time to start
+        # set pixel depth https://docs.travis-ci.com/user/gui-and-headless-browsers/
+        - /sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_00.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :00 -ac -screen 0 1280x1024x16
```
Comment (tn-exafore) [inline .travis.yml]: why this change is not applied to the osx job?

Comment (pmiettinen) [reply .travis.yml]: The original is preferred as it's more simple. On OSX it worked at least earlier. Related https://github.com/swift-nav/piksi_tools/pull/974#discussion_r254592223

**Comment by tn-exafore** [comment]
> 
> 
> a bit late but if you want to rebase and merge this seems like a pretty valuable thing to have working. @tn-exafore do you see any incompatibilities with the upgrades you are in the process of completing?

No, it would have conflicted with the main python 3 PR for linux but that is already merged.

File: .travis.yml (hunk 4 above)
Comment (tn-exafore) [reply .travis.yml]: ok

Comment (pmiettinen) [reply .travis.yml]: Looks like latest master doesn't have this problem anymore, so I'll revert this change.
